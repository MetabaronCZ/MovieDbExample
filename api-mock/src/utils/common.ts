import { Request } from 'express';

export type QueryParam = Request['query'][number];
export type Query = Record<string, QueryParam>;

const wait = (delay: number): Promise<void> => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const requestDelay = (cb: () => void): void => {
  const delay = 800 + Math.floor(400 * Math.random());
  wait(delay)
    .then(cb)
    .catch((error: unknown) => {
      console.error(error);
    });
};

// extract simple string value from Express query parameter result
export const parseQueryParam = (param: QueryParam): string => {
  const value = Array.isArray(param) ? param[0] : param;
  return 'string' === typeof value ? value : '';
};

type Entries<T extends Record<string, unknown>> = {
  readonly [K in keyof T]: [K, T[K]];
}[keyof T][];

// extract object entries, correctly typed
export const getObjectEntries = <T extends Record<string, unknown>>(
  obj: T,
): Entries<T> => {
  return Object.entries(obj) as Entries<T>;
};
