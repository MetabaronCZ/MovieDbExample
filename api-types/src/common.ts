export const perPages = [25, 50, 100] as const;
export const defaultPerPage = perPages[0];

export const sortDirections = ['ascending', 'descending'] as const;
export type SortDirection = (typeof sortDirections)[number];
