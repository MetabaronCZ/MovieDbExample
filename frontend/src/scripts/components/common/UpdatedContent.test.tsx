import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { UpdatedContent } from 'components/common/UpdatedContent';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/UpdatedContent', () => {
  it('should render standardcontent when not loading', async () => {
    const { container } = await waitForComponent(
      <UpdatedContent>CONTENT</UpdatedContent>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    expect(content).toHaveStyle({ opacity: '' });
  });

  it('should render faded content when loading', async () => {
    const { container } = await waitForComponent(
      <UpdatedContent loading>CONTENT</UpdatedContent>,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    expect(content).not.toHaveStyle({ opacity: '' });
  });
});
