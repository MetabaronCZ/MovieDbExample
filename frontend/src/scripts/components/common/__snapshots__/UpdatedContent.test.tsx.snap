// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/common/UpdatedContent should render faded content when loading 1`] = `
.c0 {
  opacity: 0.65;
  pointer-events: none;
}

<div>
  <div
    class="c0"
  >
    CONTENT
  </div>
</div>
`;

exports[`components/common/UpdatedContent should render standard content when not loading 1`] = `
<div>
  CONTENT
</div>
`;
